graphlearn_torch/csrc/cpu/random_sampler.cc

```diff
@@ -175,4 +281,51 @@ void CPURandomSampler::UniformSample(const int64_t* col_begin,
   }
 }
 
+void CPURandomSampler::WeightedRandomSample(const int64_t* col_begin,
+                                            const int64_t* col_end,
+                                            const int32_t req_num,
+                                            const float* prob_begin,
+                                            const float* prob_end,
+                                            int64_t* out_nbrs) {
+  // with replacement
+  const auto cap = col_end - col_begin;
+  if (req_num < cap) {
+    thread_local static std::random_device rd;
+    thread_local static std::mt19937 engine(rd());
+    std::discrete_distribution<> dist(prob_begin, prob_end);
```

Comment:
This function only works with integers. Here prob is float, It is best to use something like the alias method or reservoir algorithm for weighted sampling.

Reply:
As shown in the test_neighbor_sampler.py, this function works well with float. The weights are set 0.1 and 0.9 for the neighbors and the sampling frequency obeys the distribution we assumed